File: src/main/java/bdv/bigcat/viewer/atlas/opendialog/DatasetInfo.java
package bdv.bigcat.viewer.atlas.opendialog;

import java.util.stream.Stream;

import javafx.beans.property.DoubleProperty;
import javafx.beans.property.SimpleDoubleProperty;

public class DatasetInfo
{

	private final DoubleProperty[] spatialResolution = Stream.generate( SimpleDoubleProperty::new ).limit( 3 ).toArray( DoubleProperty[]::new );

	private final DoubleProperty[] spatialOffset = Stream.generate( SimpleDoubleProperty::new ).limit( 3 ).toArray( DoubleProperty[]::new );

	private final SimpleDoubleProperty min = new SimpleDoubleProperty( Double.NaN );

	private final SimpleDoubleProperty max = new SimpleDoubleProperty( Double.NaN );

	public DoubleProperty[] spatialResolutionProperties()
	{
		return this.spatialResolution;
	}

	public DoubleProperty[] spatialOffsetProperties()
	{
		return this.spatialOffset;
	}

	public DoubleProperty minProperty()
	{
		return this.min;
	}

	public DoubleProperty maxProperty()
	{
		return this.max;
	}

}

File: src/main/kotlin/org/janelia/saalfeldlab/paintera/ui/opendialog/DatasetInfo.kt
package org.janelia.saalfeldlab.paintera.ui.opendialog

import javafx.beans.property.DoubleProperty
import javafx.beans.property.SimpleDoubleProperty

class DatasetInfo {


    val spatialResolutionProperties: Array<DoubleProperty> = Array(3) { SimpleDoubleProperty() }

    val spatialOffsetProperties: Array<DoubleProperty> = Array(3) { SimpleDoubleProperty() }

    val minProperty = SimpleDoubleProperty(Double.NaN)

    val maxProperty = SimpleDoubleProperty(Double.NaN)
}
